zap/src/main/java/org/parosproxy/paros/network/HttpSender.java

```diff
@@ -237,6 +240,7 @@ public HttpSender(ConnectionParam connectionParam, boolean useGlobalState, int i
                         defaultUserAgent);
 
         setUseGlobalState(useGlobalState);
+        setUseCookies(true);
```

Comment:
There still seems to be a sync issue with opening and closing the dialog and the button state/behaviour. Setting `setUseCookies(useCookies);` here seems closer but still not 100% :cry:

Reply:
@kingthorin Weird, will test more and try to see where the problem is!

Reply:
@kingthorin should be fixed in the latest commit, the state wasn't being properly relayed from the graphical interface.

Reply:
Great, I'll test it out later tonight or in the AM.